### DISCUSSION TITLE
z ignores current directory ?

### DISCUSSION THREAD

**Discussion by hemna** [ajeetdsouza/zoxide#770] [Q&A]
I am trying to get used to zoxide and really like it.  The one major thing that I have to say is VERY annoying is that it alway defaults to the db entries first and ignores the current directory.  

The only way I can force it to cd to a current directory subdirectory is to specifiy ./<subdirectory>

For example,
  I have multiple places on my filesystem that have 'aprsd' in the name.   It's an app I developing and an app that I deploy in docker.

I have:

```
/docker/aprsd-wb4bor-11
/docker/aprsd-hemna
....

~/devel/aprsd
~/devel/aprsd/aprsd

~/tmp/aprsd
```
You get the idea.  

if I type 'z aprsd' it typically goes to ~/devel/aprsd, which is great

but once inside of ~/devel/aprsd I want to cd again to ./aprsd

The problem is, that I have an alias of cd=z
and I Want to cd into ./aprsd, but will type
cd aprsd, because aprsd is in the current directory.  it ends up doing nothing, or if my z db has another location as visited more often, it will send me back there...such as /docker/aprsd-wb4for-11, when all I wanted to do is go into the aprsd directory from the current location I was in.   z basically ignores the current directory looking for the right place to change to unless I force it to ./aprsd.

It would be nice if there was an option that I could use (and place in my alias) that defaults to looking in the current directory first for the location to change to, and if it's not there, then fallback to the database location.

**Comment by ajeetdsouza** [comment]
If there is only 1 argument (like `z aprsd`), zoxide will check the current directory before checking the database. This is the code for it: https://github.com/ajeetdsouza/zoxide/blob/94d3cba60e24139319c7d370d796967c607b54ef/templates/zsh.txt#L56-L67

How are you initializing zoxide?

**Comment by viniciusth** [comment]
I have the same issue, my setup is 
```bash
alias cd='z'
eval "$(zoxide init zsh)"
```
in my .zshrc

example of problem that I have:
```bash
~/pathA $ ls 
folderA
folderB
~/pathA $ cd folderA
~/pathB/folderA $
```

the expected / wanted behavior on my part is for it to go to `~/pathA/folderA` since it is an exact match but that doesn't happen.
is there a way for it to work like that @ajeetdsouza ?

**Comment by faberfedor** [comment]
Let me add my two cents:

# alias set to cd='z'
mkdir foo
cd f

Will read suggestions from t z database and not the local directory. Continuing...

cd fo

will give no suggestions. Nor will

cd ./fo

Every indication I see tells me zoxide is not reading the local directory.

I'm running zsh and oh-my-zsh on an M2 macbook pro.

**Reply by faberfedor** [reply]
That large 'alias set to cd is supposed to be a comment, not a heading.

**Reply by faberfedor** [reply]
Found my problem: you can set the TAB key to accept (TAB) completions XOR autosuggest-accept (read: completions or autosuggest-accept but not both). I bound autosuggest-accept to shift-TAB and it works.

Let's see how well it works with muscle memory...
